pontos/version/commands/_java.py

```diff
@@ -15,199 +15,160 @@
-        except etree.XMLSyntaxError as e:
-            raise VersionError(e) from e
+    def get_current_version(self) -> Version:
+        file_versions = self._read_versions_from_files()
 
-        self._pom_xml = pom_xml.getroot()
+        last_version = self._verify_version(file_versions)
 
-        return self._pom_xml
+        if last_version == "":
+            raise VersionError("no version found")
 
-    def get_current_version(self) -> Version:
-        """Get the current version of this project
-        In go the version is only defined within the repository
-        tags, thus we need to check git, what tag is the latest"""
-        return self._get_version_from_pom_xml()
+        return self.versioning_scheme.parse_version(last_version)
 
     def verify_version(
         self, version: Union[Literal["current"], Version, None]
     ) -> None:
-        """Verify the current version of this project"""
-        current_version = self.get_current_version()
+        file_versions = self._read_versions_from_files()
+
+        last_version = self._verify_version(file_versions)
 
-        if current_version != version:
+        if last_version != str(version):
             raise VersionError(
                 f"Provided version {version} does not match the "
-                f"current version {current_version} in "
-                f"{self.project_file_path}."
+                f"current version {last_version} "
+                f"in '{self.project_file_path}'"
             )
 
     def update_version(
         self, new_version: Version, *, force: bool = False
     ) -> VersionUpdate:
         try:
-            package_version = self.get_current_version()
-            if not force and new_version == package_version:
-                return VersionUpdate(previous=package_version, new=new_version)
+            current_version = self.get_current_version()
+            if not force and new_version == current_version:
+                return VersionUpdate(previous=current_version, new=new_version)
         except VersionError:
             # just ignore current version and override it
-            package_version = None
+            current_version = None
 
-        changed_files = [self.project_file_path]
-        self._update_pom_version(new_version=new_version)
-        self._update_properties_file(new_version=new_version)
-        self._update_swagger_config(new_version=new_version)
+        changed_files = self._update_version_files(new_version)
 
         return VersionUpdate(
-            previous=package_version,
+            previous=current_version,
             new=new_version,
             changed_files=changed_files,
         )
+
+    def _update_version_files(self, new_version) -> List[Path]:
+        config = self._load_config()
+
+        changed_files: List[Path] = []
+        for file_config in config["files"]:
+            file_path = file_config["path"]
+            with open(Path.cwd() / file_path, "r") as input_file_handle:
```

Comment:
```suggestion
            with (Path.cwd() / file_path).open("r") as input_file_handle:
```

But as far as I can see you just need the text of the file therefore you can just use
```
content = (Path.cwd() / file_path).read_text(encoding="utf8")
```